### PR TITLE
Add NetApp ONTAP user module

### PR DESCRIPTION
**Pull request by timuster**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

NetApp ONTAP user module

##### ANSIBLE VERSION

```
ansible 2.3.0
  config file = 
  configured module search path = Default w/o overrides
```

##### SUMMARY

Adding na_cdot_user.py, a module to manage NetApp ONTAP users. The design was based on the existing NetApp E-Series modules.

```

```

### CODE REVIEW

**Comment by hulquest** [comment]
LGTM.
shipit

File: lib/ansible/modules/storage/netapp/na_cdot_user.py [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+  application:
+    description:
+    - Applications to grant access to.
+    required: true
+    choices: ['console', 'http','ontapi','rsh','snmp','sp','ssh','telnet']
+
+  authentication_method:
+    description:
+    - Authentication method for the application.
+    - Not all authentication methods are valid for an application. Valid authentication methods for each application are as denoted in I(authentication_choices_description).
```
Comment (mattclay) [inline lib/ansible/modules/storage/netapp/na_cdot_user.py]: CI failure due to:

```
2017-02-10 20:57:26 ERROR: PEP 8: lib/ansible/modules/storage/netapp/na_cdot_user.py:59:161: E501 line too long (173 > 160 characters) (current)
```

**Comment by hulquest** [comment]
shipit

**Comment by lmprice** [comment]
shipit

**Pull request merged**